### ISSUE TITLE
Performance degradation between rc1 to rc2 of 2.0

### ISSUE THREAD

**Issue by thaumos** [ansible/ansible#13510]
Issue Type: Bug Report

Ansible Version: 2.0.0-rc2

``` bash
ansible 2.0.0 (2.0.rc2 cc98528ecb) last updated 2015/12/09 15:12:05 (GMT -700)
  lib/ansible/modules/core: (detached HEAD 29a83704a8) last updated 2015/12/10 17:03:45 (GMT -700)
  lib/ansible/modules/extras: (detached HEAD 2903021943) last updated 2015/12/10 17:03:45 (GMT -700)
  config file =
  configured module search path = Default w/o overrides
```

Ansible Configuration:

N/A, standard install.

Environment:

N/A

Summary:

Steps To Reproduce:

``` bash
time ansible-playbook repro.yml -f 50 -k --ask-become-pass -vvvv (per customer report)
```

``` yaml
- name: performance_repro
  hosts: all
  gather_facts: False
  tasks:
     - name: get uptime
       shell: uptime
       delegate_to: localhost
```

Expected Results:

Timing to be == | < 2.0rc1.  Timing increased. See below results.

Actual Results:

``` bash
1.9.4-1

Run1
real    1m9.039s
user    4m46.018s
sys 2m43.012s

Run2
real    1m8.511s
user    4m46.994s
sys 2m43.551s

Run3
real    1m7.711s
user    4m33.638s
sys 2m34.662s

Run4
real    1m7.884s
user    4m44.910s
sys 2m44.113s

Run5
real    1m7.590s
user    4m46.091s
sys 2m42.781s

2.0.0-rc1

Run1

real    8m35.138s
user    31m58.551s
sys 30m22.538s

Run2
real    8m59.590s
user    33m57.321s
sys 32m25.790s

Run3
real    9m1.210s
user    34m10.220s
sys 32m46.111s

Run4
real    6m24.033s
user    24m59.184s
sys 22m27.699s

Run5
real    7m22.810s
user    28m23.683s
sys 26m9.295s

2.0.0-rc2
Run1
real    11m33.463s
user    42m37.801s
sys 41m56.344s

Run2
real    9m43.599s
user    35m28.669s
sys 33m26.751s

Run3
real    9m21.667s
user    35m13.703s
sys 33m1.289s

Run4
real    8m3.731s
user    31m18.057s
sys 28m39.177s

Run5
real    9m3.006s
user    35m0.188s
sys 33m9.419s

DEVEL c402325085

Run1
real    9m22.027s
user    35m24.473s
sys 33m38.588s

Run2
real    8m37.824s
user    33m26.592s
sys 31m15.707s

Run3
real    8m50.095s
user    34m43.036s
sys 32m18.526s

Run4
real    8m49.278s
user    34m27.405s
sys 32m23.278s

Run5
real    8m46.609s
user    34m19.224s
sys 32m12.463s
```

**Comment by Yannig** [comment]
Can you provide any information regarding you inventory?

**Comment by thaumos** [comment]
3000 hosts, 30 groups.

**Comment by jimi-c** [comment]
The `fork_late` has been merged into both devel and stable-2.0, so based on the other numbers you shot me off thread I'm going to go ahead and close this.

Thanks!
